bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java

```diff
@@ -0,0 +1,724 @@
+                } else if (command.equals(PercentType.HUNDRED)) {
+                    command = UpDownType.DOWN;
+                }
+                if (command == UpDownType.UP) {
+                    action = BondDeviceAction.OPEN;
+                } else if (command == UpDownType.DOWN) {
+                    action = BondDeviceAction.CLOSE;
+                } else if (command == StopMoveType.STOP) {
+                    action = BondDeviceAction.HOLD;
+                }
+                if (action != null) {
+                    api.executeDeviceAction(config.deviceId, action, null);
+                }
+                break;
+
+            default:
+                logger.info("Command {} on unknown channel {}, {}", command.toFullString(), channelUID.getId(),
+                        channelUID.toString());
+                return;
+        }
+    }
+
+    private void enableUpLight() {
+        Objects.requireNonNull(api).executeDeviceAction(config.deviceId, BondDeviceAction.TURN_UP_LIGHT_ON, null);
+    }
+
+    private void enableDownLight() {
+        Objects.requireNonNull(api).executeDeviceAction(config.deviceId, BondDeviceAction.TURN_DOWN_LIGHT_ON, null);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(BondDeviceConfiguration.class);
+        logger.trace("Starting initialization for Bond device with device id {}.", config.deviceId);
+        fullyInitialized = false;
+        disposed = false;
+
+        // set the thing status to UNKNOWN temporarily
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(this::initializeThing);
+    }
+
+    @Override
+    public synchronized void dispose() {
+        logger.debug("Disposing thing handler for {}.", this.getThing().getUID());
+        // Mark handler as disposed as soon as possible to halt updates
+        disposed = true;
+        fullyInitialized = false;
+
+        final ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+        }
+        this.pollingJob = null;
+    }
+
+    private void initializeThing() {
+        if (!getBridgeAndAPI()) {
+            return;
+        }
+        BondHttpApi api = this.api;
+        if (api == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/offline.comm-error.no-api");
+            return;
+        }
+
+        try {
+            logger.trace("Getting device information for {} ({})", config.deviceId, this.getThing().getLabel());
+            deviceInfo = api.getDevice(config.deviceId);
+            logger.trace("Getting device properties for {} ({})", config.deviceId, this.getThing().getLabel());
+            deviceProperties = api.getDeviceProperties(config.deviceId);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```

Comment:
When this exception is thrown during initialization, status will be set to COMMUNICATION_ERROR. After this, it will be set to CONFIGURATION_ERROR in line 461. And no polling job will be started. Intended? And how will it recover?

Reply:
Okay, I see now that `initializeThing` is also called from `bridgeStatusChanged`, so perhaps only the status changes right after each other needs fixing.

Reply:
Yeah, I think there are quite a bit of double-set-status, especially when a device gets set offline, it sets the bridge offline, and then I suspect it ping-pongs back and sets the device offline again, with a different error. I already eliminated one set earlier. I'll look and see if there are more I can get rid of.